src/NLog/Internal/ObjectPools/ReusableBuilderCreator.cs

```diff
@@ -41,7 +41,13 @@ namespace NLog.Internal
     internal sealed class ReusableBuilderCreator : ReusableObjectCreator<StringBuilder>
     {
         public ReusableBuilderCreator()
-            : base(128, (cap) => new StringBuilder(cap), (sb) => { sb.ClearBuilder(); })
+            : base(128, (cap) => new StringBuilder(cap), (capacity, sb) =>
+            {
+                if (sb.Capacity > capacity)
```

Comment:
Can this be the logic from `StringBuilderPool` extracted into a method:
```c#
                if (sb.Length > maxCapacity)
                {
                    sb.Remove(0, stringBuilder.Length - 1);  // Attempt soft clear that skips re-allocation
                    if (sb.Capacity > maxCapacity)
                    {
                        sb.Capacity = maxCapacity / 2;
                    }
                }
```

Reply:
Can you share what remove all characters except one, better than doing Clean and setting Capacity? I couldn't find any confirmation of this.

Reply:
The StringBuilder works kind of like your `BufferPoolStream` where it chains together multiple array into a stream. The StringBuilder Capacity comes from the total size of its internal arrays. When removing all characters except one, then  StringBuilder should attempt to hold on to its initial array, and lets go of the other char-array for the garbage collector to eat.

Reply:
for me really many tests failed. I had this before my changes. I don't know what the reason is.
![image](https://github.com/user-attachments/assets/debd7fc3-d338-4655-aaa4-2db878ad5576)

Reply:
Maybe your TestRunner are running the tests in parallel? The original NLog unit-tests are depending on the global singleton `LogManager`, which causes side-effects when running in parallel. I have tried to fix many of the unit-tests with NLog v5, but still a long road ahead. Hopefully with NLog v6, then then all unit-test will be fixed.

Any help with improving the NLog unit-test is most wellcome.